packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,258 @@
+  `,
+
+  large: css`
+    ${mq({
+      width: ['720px', '720px', '960px'],
+    })}
+  `,
+};
+
+const closeButton = css`
+  color: ${uiColors.gray.dark1};
+  position: absolute;
+  cursor: pointer;
+  right: 12px;
+  top: 12px;
+`;
+
+const escapeKey = 27;
+
+interface ModalProps {
+  /**
+   * Content that will appear inside of the Modal component.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Determines the active state of the modal
+   *
+   * default: `false`
+   */
+  active: boolean;
+
+  /**
+   * Specifies the size of the Modal.
+   *
+   * default: `default`
+   */
+  size?: ModalSize;
+
+  /**
+   * Callback to change the active state of the Modal.
+   *
+   */
+  setActive?: (
+    active: boolean,
+  ) => void | React.Dispatch<SetStateAction<boolean>>;
+
+  /**
+   * Callback to determine whether or not Modal should close when user tries to close it.
+   *
+   */
+  shouldClose?: () => boolean;
+
+  /**
+   * className applied to root overlay div.
+   *
+   */
+  className?: string;
+}
+
+/**
+ * # Modal
+ *
+ *  Modals place content on top of main window.
+ *
+```
+<Modal
+  active
+  size="large"
+  setActive={setActive}
+  shouldClose={() => console.log('Modal is closing now!')}
+  >
+  Modal content!
+</Modal>
+```
+ * @param props.active Boolean to describe whether or not Modal is active.
+ * @param props.size String to determine size of Modal. ['small', 'default', 'large']
+ * @param props.setActive Callback to change the active state of Modal.
+ * @param props.children Content to appear inside of Modal container.
+ * @param props.shouldClose Callback to determine whether or not Modal should close when user tries to close it.
+ * @param props.className className applied to overlay div.
+ *
+ */
+function Modal({
+  active = false,
+  size = ModalSize.Default,
+  setActive,
+  children,
+  shouldClose,
+  className,
+  ...rest
+}: ModalProps) {
+  if (!active) {
+    return null;
+  }
+
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = useCallback(() => {
+    // Don't close modal if shouldClose returns false or no setActive callback is passed
+    if (!setActive || (shouldClose && !shouldClose())) {
+      return;
+    }
+
+    setActive(false);
+  }, [setActive, shouldClose]);
+
+  const handleDocumentClick = (e: React.SyntheticEvent) => {
+    if (!contentRef.current) {
+      return;
+    }
+
+    if (e.target !== contentRef.current) {
+      handleClose();
+    }
+  };
+
+  const handleEscape = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.keyCode === escapeKey) {
+        handleClose();
+      }
+    },
+    [handleClose],
+  );
+
+  useEventListener('keydown', handleEscape, {
+    options: { once: true },
+  });
+
+  return (
+    <Portal>
+      <div {...rest} onClick={handleDocumentClick} className={mainContainer}>
+        <div className={scrollContainer}>
+          <div
+            className={cx(modalContentStyle, modalSizes[size], className)}
+            tabIndex={-1}
```

Comment:
Generally I prefer static property values to be declared as strings to mirror HTML.

Reply:
Additionally, the modal still appears to be focusable when testing it.

Reply:
Can you explain what you mean by static property values being declared as strings, sorry!

Reply:
This value will always be `-1`, so we can simply say `tabIndex='-1'` for the property instead of indicating it needs to be evaluated as JS with `{-1}`

Reply:
Typescript complains and says that tabIndex should be a number or undefined when you pass a string

Reply:
Ahh okay. 👍